needy/projects/make.py

```diff
@@ -5,6 +5,18 @@
 from .. import project
 
 
+def get_make_jobs_args(project):
+    jobs = project.needy.jobs()
+    if project.configuration('max-jobs') is not None:
+        jobs = min(jobs, project.configuration('max-jobs'))
```

Comment:
it looks like every `Project` subclass is going to need to use these few lines. put a `max_concurrency` method in `Project`

then you can just add a one-liner like this to each subclass:
`make_args += ['-j', str(self.max_concurrency())] if self.max_concurrency() > 0 else ['-j']`